### PR TITLE
Add thunk / reducer for deleting a comment

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-code-manager/issues/1121

### CODE REVIEW

File: src/reducers/comments.tsx [hunk 1]
```diff
@@ -124,6 +136,9 @@ export const actions = {
   beginComment: createAction('BEGIN_COMMENT', (resolve) => {
     return (payload: CommonPayload) => resolve(payload);
   }),
+  beginDeleteComment: createAction('BEGIN_DELETE_COMMENT', (resolve) => {
```
Comment (kumar303) [inline src/reducers/comments.tsx]: This means that the *actual* delete API request has begun.

File: src/reducers/comments.tsx [hunk 2]
```diff
@@ -134,13 +149,19 @@ export const actions = {
     return (payload: CommonPayload & { pendingCommentText: string | null }) =>
       resolve(payload);
   }),
+  considerDeleteComment: createAction('CONSIDER_DELETE_COMMENT', (resolve) => {
```
Comment (kumar303) [inline src/reducers/comments.tsx]: This means that the UI is asking *Do you really want to delete ...?*

File: src/reducers/comments.tsx [hunk 3]
```diff
@@ -209,6 +240,38 @@ export const fetchAndLoadComments = ({
...
+  addonId: number;
+  commentId: number;
+  versionId: number;
+}): ThunkActionCreator => {
+  return async (dispatch, getState) => {
+    const { api: apiState } = getState();
+
+    dispatch(actions.beginDeleteComment({ commentId }));
+
+    const response = await _requestCommentDeletion({
```
Comment (willdurand) [inline src/reducers/comments.tsx]: Nit: I understand why it is named like this, but I find this new name "far" from the original name.

Comment (kumar303) [reply src/reducers/comments.tsx]: And that makes the code harder to follow? Or something else? I changed the alias to `apiDeleteComment` which is closer to the original name but the code is a bit less readable to me. In any case, no biggie -- I don't mind changing it.

**Pull request merged**